### PR TITLE
Document few changes for Matomo 4

### PR DESCRIPTION
**Pull request by tsteur**
refs https://github.com/matomo-org/matomo/issues/8415 
refs https://github.com/matomo-org/matomo/issues/12420

Updated some docs based on the changelog and also updated some domains, piwik.js and matomo.js mentions etc.

### CODE REVIEW

**Comment by tsteur** [comment]
@sgiehl  or @Findus23 if someone could have a very quick look that be great.

File: docs/4.x/ab-tests/browser.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ _paq.push(['AbTesting::create', {
       }
       return false;
   },
-  matomoTracker: Matomo.getAsyncTracker(piwikUrl, piwikSiteId),
+  piwikTracker: Matomo.getAsyncTracker(matomoUrl, matomoSiteId),
```
Comment (sgiehl) [inline docs/4.x/ab-tests/browser.md:110]: Guess the `piwikTracker` option hasn't been renamed in the plugin yet?

**Review by sgiehl — APPROVED** [review]
LGTM

**Review by Findus23 — APPROVED** [review]
Didn't see anything weird when scrolling through. Maybe once we have replaced all piwik occurrences, we could remove the automatic replacement as I think it still breaks some pages.

File: docs/4.x/ab-tests/browser.md (hunk 1 above)
Comment (tsteur) [reply docs/4.x/ab-tests/browser.md:110]: exactly, will change the premium feature docs once we renamed things there

**Pull request merged**